libfwupdplugin/fu-common.c

```diff
@@ -1923,3 +1929,37 @@ fu_byte_array_append_uint32 (GByteArray *array, guint32 data, FuEndianType endia
 	fu_common_write_uint32 (buf, data, endian);
 	g_byte_array_append (array, buf, sizeof(buf));
 }
+
+/**
+ * fu_common_kernel_locked_down:
+ *
+ * Determines if kernel lockdown in effect
+ *
+ * Since: 1.3.8
+ **/
+gboolean
+fu_common_kernel_locked_down (void)
+{
+#ifndef _WIN32
+	gsize len = 0;
+	g_autofree gchar *dir = fu_common_get_path (FU_PATH_KIND_SYSFSDIR_SECURITY);
+	g_autofree gchar *fname = g_build_filename (dir, "lockdown", NULL);
+	g_autofree gchar *data = NULL;
+	g_auto(GStrv) options = NULL;
+
+	if (!g_file_test (fname, G_FILE_TEST_EXISTS))
+		return FALSE;
+	if (!g_file_get_contents (fname, &data, &len, NULL))
+		return FALSE;
+	if (len < 1)
+		return FALSE;
+	options = g_strsplit (data, " ", -1);
+	for (guint i = 0; options[i] != NULL; i++) {
+		if (g_strcmp0 (options[i], "[none]") == 0)
```

Comment:
interesting, are the square brackets meaning inverted?

Reply:
Here is what it looks like:
```
$ sudo cat /sys/kernel/security/lockdown 
none integrity [confidentiality]
```

I was thinking that detecting the `none` case makes most sense in case more options are added later.

Reply:
I'll admit I'm a bit confused what value means what.

Reply:
I've seen this nomenclature in the kernel sysfs files previously, and actually used it myself in a driver I wrote.

The thing in the brackets is what's picked.  The other things are possible options.  So rather than two sysfs files that are:
`current_value` and 
`possible_values`

You just read one and can tell all the possible values and which one is active.

Reply:
TIL, thanks.